// @bilibili/b-utils-core/es/index.js
import * as consts from './consts';
export { consts };
import version from './version';
export { default as version } from './version';
import isPhone from './isPhone';
export { default as isPhone } from './isPhone';
import isEmail from './isEmail';
export { default as isEmail } from './isEmail';
import isIdCard from './isIdCard';
export { default as isIdCard } from './isIdCard';
import isUrl from './isUrl';
export { default as isUrl } from './isUrl';
import isIpv4 from './isIpv4';
export { default as isIpv4 } from './isIpv4';
import isCnChar from './isCnChar';
export { default as isCnChar } from './isCnChar';
import isDbChar from './isDbChar';
export { default as isDbChar } from './isDbChar';
import isBase64 from './isBase64';
export { default as isBase64 } from './isBase64';
import isArr from './isArr';
export { default as isArr } from './isArr';
import isObj from './isObj';
export { default as isObj } from './isObj';
import isFun from './isFun';
export { default as isFun } from './isFun';
import isStr from './isStr';
export { default as isStr } from './isStr';
import isBol from './isBol';
export { default as isBol } from './isBol';
import isNum from './isNum';
export { default as isNum } from './isNum';
import isNul from './isNul';
export { default as isNul } from './isNul';
import isUdf from './isUdf';
export { default as isUdf } from './isUdf';
import isUdfNul from './isUdfNul';
export { default as isUdfNul } from './isUdfNul';
import toRowType from './toRowType';
export { default as toRowType } from './toRowType';
import array from './array';
export { default as array } from './array';
import toArray from './toArray';
export { default as toArray } from './toArray';
import tulpeArray from './tulpeArray';
export { default as tulpeArray } from './tulpeArray';
import unique from './unique';
export { default as unique } from './unique';
import shuffle from './shuffle';
export { default as shuffle } from './shuffle';
import flatten from './flatten';
export { default as flatten } from './flatten';
import includes from './includes';
export { default as includes } from './includes';
import findIndex from './findIndex';
export { default as findIndex } from './findIndex';
import get from './get';
export { default as get } from './get';
import pick from './pick';
export { default as pick } from './pick';
import omit from './omit';
export { default as omit } from './omit';
import deepCopy from './deepCopy';
export { default as deepCopy } from './deepCopy';
import isDeepEqual from './isDeepEqual';
export { default as isDeepEqual } from './isDeepEqual';
import pretty from './pretty';
export { default as pretty } from './pretty';
import debounce from './debounce';
export { default as debounce } from './debounce';
import throttle from './throttle';
export { default as throttle } from './throttle';
import randomId from './randomId';
export { default as randomId } from './randomId';
import randomNum from './randomNum';
export { default as randomNum } from './randomNum';
import randomColor from './randomColor';
export { default as randomColor } from './randomColor';
import getStringHash from './getStringHash';
export { default as getStringHash } from './getStringHash';
import compareVersion from './compareVersion';
export { default as compareVersion } from './compareVersion';
import parseQueryString from './parseQueryString';
export { default as parseQueryString } from './parseQueryString';
import parseQs from './parseQs';
export { default as parseQs } from './parseQs';
import stringifyQueryString from './stringifyQueryString';
export { default as stringifyQueryString } from './stringifyQueryString';
import stringifyQs from './stringifyQs';
export { default as stringifyQs } from './stringifyQs';
import escapeHtml from './escapeHtml';
export { default as escapeHtml } from './escapeHtml';
import encodeHtml from './encodeHtml';
export { default as encodeHtml } from './encodeHtml';
import unescapeHtml from './unescapeHtml';
export { default as unescapeHtml } from './unescapeHtml';
import decodeHtml from './decodeHtml';
export { default as decodeHtml } from './decodeHtml';
import formatNum from './formatNum';
export { default as formatNum } from './formatNum';
import formatMoney from './formatMoney';
export { default as formatMoney } from './formatMoney';
import cutString from './cutString';
export { default as cutString } from './cutString';
import maskString from './maskString';
export { default as maskString } from './maskString';
import capitalize from './capitalize';
export { default as capitalize } from './capitalize';
import outOfNumShowPlus from './outOfNumShowPlus';
export { default as outOfNumShowPlus } from './outOfNumShowPlus';
import trimReLimits from './trimReLimits';
export { default as trimReLimits } from './trimReLimits';
import CountDown from './CountDown';
export { default as CountDown } from './CountDown';
import formatToDate from './formatToDate';
export { default as formatToDate } from './formatToDate';
import formatDate from './formatDate';
export { default as formatDate } from './formatDate';
import formatDateTime from './formatDateTime';
export { default as formatDateTime } from './formatDateTime';
import formatDuration from './formatDuration';
export { default as formatDuration } from './formatDuration';
import formatTime from './formatTime';
export { default as formatTime } from './formatTime';
import formatPassedTime from './formatPassedTime';
export { default as formatPassedTime } from './formatPassedTime';
import getAgeByBirth from './getAgeByBirth';
export { default as getAgeByBirth } from './getAgeByBirth';
import getAgeByIdCard from './getAgeByIdCard';
export { default as getAgeByIdCard } from './getAgeByIdCard';
import getDayStartTime from './getDayStartTime';
export { default as getDayStartTime } from './getDayStartTime';
import getDayEndTime from './getDayEndTime';
export { default as getDayEndTime } from './getDayEndTime';
import getDayOfYear from './getDayOfYear';
export { default as getDayOfYear } from './getDayOfYear';
import getWeekOfYear from './getWeekOfYear';
export { default as getWeekOfYear } from './getWeekOfYear';
import getMonthDays from './getMonthDays';
export { default as getMonthDays } from './getMonthDays';
import getDaysBetween from './getDaysBetween';
export { default as getDaysBetween } from './getDaysBetween';
import isLeapYear from './isLeapYear';
export { default as isLeapYear } from './isLeapYear';
import PageParams from './PageParams';
export { default as PageParams } from './PageParams';
import createHttp from './createHttp';
export { default as createHttp } from './createHttp';
import toHttps from './toHttps';
export { default as toHttps } from './toHttps';
import urlToHttps from './urlToHttps';
export { default as urlToHttps } from './urlToHttps';
import trimHttp from './trimHttp';
export { default as trimHttp } from './trimHttp';
import urlTrimHttp from './urlTrimHttp';
export { default as urlTrimHttp } from './urlTrimHttp';
import isSupportWebp from './isSupportWebp';
export { default as isSupportWebp } from './isSupportWebp';
import formatImg from './formatImg';
export { default as formatImg } from './formatImg';
import formatImgSrc from './formatImgSrc';
export { default as formatImgSrc } from './formatImgSrc';
import onImagesLoad from './onImagesLoad';
export { default as onImagesLoad } from './onImagesLoad';
import getCookie from './getCookie';
export { default as getCookie } from './getCookie';
import setCookie from './setCookie';
export { default as setCookie } from './setCookie';
import removeCookie from './removeCookie';
export { default as removeCookie } from './removeCookie';
import getCache from './getCache';
export { default as getCache } from './getCache';
import setCache from './setCache';
export { default as setCache } from './setCache';
import removeCache from './removeCache';
export { default as removeCache } from './removeCache';
import getLocalStorage from './getLocalStorage';
export { default as getLocalStorage } from './getLocalStorage';
import setLocalStorage from './setLocalStorage';
export { default as setLocalStorage } from './setLocalStorage';
import removeLocalStorage from './removeLocalStorage';
export { default as removeLocalStorage } from './removeLocalStorage';
import getSessionStorage from './getSessionStorage';
export { default as getSessionStorage } from './getSessionStorage';
import setSessionStorage from './setSessionStorage';
export { default as setSessionStorage } from './setSessionStorage';
import removeSessionStorage from './removeSessionStorage';
export { default as removeSessionStorage } from './removeSessionStorage';
import isMobile from './isMobile';
export { default as isMobile } from './isMobile';
import isDesktop from './isDesktop';
export { default as isDesktop } from './isDesktop';
import isIos from './isIos';
export { default as isIos } from './isIos';
import isAndroid from './isAndroid';
export { default as isAndroid } from './isAndroid';
import isSSR from './isSSR';
export { default as isSSR } from './isSSR';
import isInBrowser from './isInBrowser';
export { default as isInBrowser } from './isInBrowser';
import isZhTw from './isZhTw';
export { default as isZhTw } from './isZhTw';
import isTouchScreen from './isTouchScreen';
export { default as isTouchScreen } from './isTouchScreen';
import getBrowserInfo from './getBrowserInfo';
export { default as getBrowserInfo } from './getBrowserInfo';
import getPlatformInfo from './getPlatformInfo';
export { default as getPlatformInfo } from './getPlatformInfo';
import getPerformanceInfo from './getPerformanceInfo';
export { default as getPerformanceInfo } from './getPerformanceInfo';
import copyText from './copyText';
export { default as copyText } from './copyText';
import loadScript from './loadScript';
export { default as loadScript } from './loadScript';
import insertScript from './insertScript';
export { default as insertScript } from './insertScript';
import loadCss from './loadCss';
export { default as loadCss } from './loadCss';
import insertCss from './insertCss';
export { default as insertCss } from './insertCss';
import isFullscreen from './isFullscreen';
export { default as isFullscreen } from './isFullscreen';
import requestFullscreen from './requestFullscreen';
export { default as requestFullscreen } from './requestFullscreen';
import exitFullscreen from './exitFullscreen';
export { default as exitFullscreen } from './exitFullscreen';
import scrollTo from './scrollTo';
export { default as scrollTo } from './scrollTo';
import scrollToTop from './scrollToTop';
export { default as scrollToTop } from './scrollToTop';
import scrollToBottom from './scrollToBottom';
export { default as scrollToBottom } from './scrollToBottom';
import getScrollPosition from './getScrollPosition';
export { default as getScrollPosition } from './getScrollPosition';
import getClientSize from './getClientSize';
export { default as getClientSize } from './getClientSize';
import getPageSize from './getPageSize';
export { default as getPageSize } from './getPageSize';
import getVideoSizeRatio from './getVideoSizeRatio';
export { default as getVideoSizeRatio } from './getVideoSizeRatio';
import setToFavorite from './setToFavorite';
export { default as setToFavorite } from './setToFavorite';
import requestAnimationFrame from './requestAnimationFrame';
export { default as requestAnimationFrame } from './requestAnimationFrame';
import getScreenPosition from './getScreenPosition';
export { default as getScreenPosition } from './getScreenPosition';
import openWindow from './openWindow';
export { default as openWindow } from './openWindow';
import onWindowResize from './onWindowResize';
export { default as onWindowResize } from './onWindowResize';
import offWindowResize from './offWindowResize';
export { default as offWindowResize } from './offWindowResize';
import dispatchCustomEvent from './dispatchCustomEvent';
export { default as dispatchCustomEvent } from './dispatchCustomEvent';
import isHTMLElement from './isHTMLElement';
export { default as isHTMLElement } from './isHTMLElement';
import isElementMatch from './isElementMatch';
export { default as isElementMatch } from './isElementMatch';
import getClosestElement from './getClosestElement';
export { default as getClosestElement } from './getClosestElement';
import getMatchedWrapElement from './getMatchedWrapElement';
export { default as getMatchedWrapElement } from './getMatchedWrapElement';
import removeDomElement from './removeDomElement';
export { default as removeDomElement } from './removeDomElement';
import setCursorPosition from './setCursorPosition';
export { default as setCursorPosition } from './setCursorPosition';
import hasClass from './hasClass';
export { default as hasClass } from './hasClass';
import addClass from './addClass';
export { default as addClass } from './addClass';
import removeClass from './removeClass';
export { default as removeClass } from './removeClass';
import base64ToBlob from './base64ToBlob';
export { default as base64ToBlob } from './base64ToBlob';
import base64ToFile from './base64ToFile';
export { default as base64ToFile } from './base64ToFile';
import base64ToUnit8Array from './base64ToUnit8Array';
export { default as base64ToUnit8Array } from './base64ToUnit8Array';
import canvasVoidBlur from './canvasVoidBlur';
export { default as canvasVoidBlur } from './canvasVoidBlur';
import downloadFile from './downloadFile';
export { default as downloadFile } from './downloadFile';
import fileOrBlobToBase64 from './fileOrBlobToBase64';
export { default as fileOrBlobToBase64 } from './fileOrBlobToBase64';
import imageToBase64 from './imageToBase64';
export { default as imageToBase64 } from './imageToBase64';
import allCustomReport from './allCustomReport';
export { default as allCustomReport } from './allCustomReport';

// 常量配置
var index = {
  consts: consts,
  version: version,
  isPhone: isPhone,
  isEmail: isEmail,
  isIdCard: isIdCard,
  isUrl: isUrl,
  isIpv4: isIpv4,
  isCnChar: isCnChar,
  isDbChar: isDbChar,
  isBase64: isBase64,
  isArr: isArr,
  isObj: isObj,
  isFun: isFun,
  isStr: isStr,
  isBol: isBol,
  isNum: isNum,
  isNul: isNul,
  isUdf: isUdf,
  isUdfNul: isUdfNul,
  toRowType: toRowType,
  array: array,
  toArray: toArray,
  tulpeArray: tulpeArray,
  unique: unique,
  shuffle: shuffle,
  flatten: flatten,
  includes: includes,
  findIndex: findIndex,
  get: get,
  pick: pick,
  omit: omit,
  deepCopy: deepCopy,
  isDeepEqual: isDeepEqual,
  pretty: pretty,
  debounce: debounce,
  throttle: throttle,
  randomId: randomId,
  randomNum: randomNum,
  randomColor: randomColor,
  getStringHash: getStringHash,
  compareVersion: compareVersion,
  parseQueryString: parseQueryString,
  parseQs: parseQs,
  stringifyQueryString: stringifyQueryString,
  stringifyQs: stringifyQs,
  escapeHtml: escapeHtml,
  encodeHtml: encodeHtml,
  unescapeHtml: unescapeHtml,
  decodeHtml: decodeHtml,
  formatNum: formatNum,
  formatMoney: formatMoney,
  cutString: cutString,
  maskString: maskString,
  capitalize: capitalize,
  outOfNumShowPlus: outOfNumShowPlus,
  trimReLimits: trimReLimits,
  CountDown: CountDown,
  formatToDate: formatToDate,
  formatDate: formatDate,
  formatDateTime: formatDateTime,
  formatDuration: formatDuration,
  formatTime: formatTime,
  formatPassedTime: formatPassedTime,
  getAgeByBirth: getAgeByBirth,
  getAgeByIdCard: getAgeByIdCard,
  getDayStartTime: getDayStartTime,
  getDayEndTime: getDayEndTime,
  getDayOfYear: getDayOfYear,
  getWeekOfYear: getWeekOfYear,
  getMonthDays: getMonthDays,
  getDaysBetween: getDaysBetween,
  isLeapYear: isLeapYear,
  PageParams: PageParams,
  createHttp: createHttp,
  toHttps: toHttps,
  urlToHttps: urlToHttps,
  trimHttp: trimHttp,
  urlTrimHttp: urlTrimHttp,
  isSupportWebp: isSupportWebp,
  formatImg: formatImg,
  formatImgSrc: formatImgSrc,
  onImagesLoad: onImagesLoad,
  getCookie: getCookie,
  setCookie: setCookie,
  removeCookie: removeCookie,
  getCache: getCache,
  setCache: setCache,
  removeCache: removeCache,
  getLocalStorage: getLocalStorage,
  setLocalStorage: setLocalStorage,
  removeLocalStorage: removeLocalStorage,
  getSessionStorage: getSessionStorage,
  setSessionStorage: setSessionStorage,
  removeSessionStorage: removeSessionStorage,
  isMobile: isMobile,
  isDesktop: isDesktop,
  isIos: isIos,
  isAndroid: isAndroid,
  isSSR: isSSR,
  isInBrowser: isInBrowser,
  isZhTw: isZhTw,
  isTouchScreen: isTouchScreen,
  getBrowserInfo: getBrowserInfo,
  getPlatformInfo: getPlatformInfo,
  getPerformanceInfo: getPerformanceInfo,
  copyText: copyText,
  loadScript: loadScript,
  insertScript: insertScript,
  loadCss: loadCss,
  insertCss: insertCss,
  isFullscreen: isFullscreen,
  requestFullscreen: requestFullscreen,
  exitFullscreen: exitFullscreen,
  scrollTo: scrollTo,
  scrollToTop: scrollToTop,
  scrollToBottom: scrollToBottom,
  getScrollPosition: getScrollPosition,
  getClientSize: getClientSize,
  getPageSize: getPageSize,
  getVideoSizeRatio: getVideoSizeRatio,
  setToFavorite: setToFavorite,
  requestAnimationFrame: requestAnimationFrame,
  getScreenPosition: getScreenPosition,
  openWindow: openWindow,
  onWindowResize: onWindowResize,
  offWindowResize: offWindowResize,
  dispatchCustomEvent: dispatchCustomEvent,
  isHTMLElement: isHTMLElement,
  isElementMatch: isElementMatch,
  getClosestElement: getClosestElement,
  getMatchedWrapElement: getMatchedWrapElement,
  removeDomElement: removeDomElement,
  setCursorPosition: setCursorPosition,
  hasClass: hasClass,
  addClass: addClass,
  removeClass: removeClass,
  base64ToBlob: base64ToBlob,
  base64ToFile: base64ToFile,
  base64ToUnit8Array: base64ToUnit8Array,
  canvasVoidBlur: canvasVoidBlur,
  downloadFile: downloadFile,
  fileOrBlobToBase64: fileOrBlobToBase64,
  imageToBase64: imageToBase64,
  allCustomReport: allCustomReport,
};

export default index;
