// @bilibili/b-utils-core/lib/consts.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

/**
 * 约定常量
 */
// 缓存对象名称
var BUTILS_CACHE = '__butils_cache';

// 缓存样式名称
var CACHED_STYLES = '__butils_styles';

// 缓存脚本名称
var CACHED_SCRIPTS = '__butils_scripts';

/**
 * 符号设定
 */
// ID限定字符
var ID_CHAR_SET = {
  number: '0123456789',
  letter: 'abcdefghijklmnopqrstuvwxyz',
  special: '~`!@#$%^&*()-_+=[]{};:"\',<.>/?'
};

// 转义符号列表
var HTML_ESCAPES = {
  '&': '&amp;',
  '<': '&lt;',
  '>': '&gt;',
  '"': '&quot;',
  "'": '&#39;',
  '`': '&#96;'
};

// 反转义符号列表
var HTML_UNESCAPES = {
  '&amp;': '&',
  '&lt;': '<',
  '&gt;': '>',
  '&quot;': '"',
  '&#39;': "'",
  '&#96;': '`'
};

/**
 * 时间常量
 */
// 时间-一天的毫秒数：86400000 = 1000 * 60 * 60 * 24
var DAY_TIME = 864e5;

// 时间-一天的毫秒数减一
var DAY_TIME_1 = 86399999;

/**
 * 常用正则
 */
// 正则-手机号码
var RE_PHONE = /^(\+?0?86-?)?1[3456789]\d{9}$/;

// 正则-座机号码
var RE_TEL = /^((\d{3,4})|\d{3,4}-)?\d{7,8}?$/;

// 正则-邮箱
var RE_EMAIL = /^\w+([-+.]\w+)*@\w+([-.]\w+)*\.\w+([-.]\w+)*$/;

// 正则-邮政编码
var RE_ZIP_CODE = /^[1-9]\d{5}(?!\d)$/;

// 正则-腾讯QQ号
var RE_QQ = /^[1-9][0-9]{4,}$/;

// 正则-一代身份证号码15位
var RE_ID_CARD_V1 = /^[1-9]\d{7}((0\d)|(1[0-2]))(([0|1|2]\d)|3[0-1])\d{3}$/;

// 正则-二代身份证号码18位
var RE_ID_CARD_V2 = /^[1-9]\d{5}[1-9]\d{3}((0\d)|(1[0-2]))(([0|1|2]\d)|3[0-1])((\d{4})|\d{3}[Xx])$/;

// 正则-身份证号码
var RE_ID_CARD = /^[1-9]\d{7}((0\d)|(1[0-2]))(([0|1|2]\d)|3[0-1])\d{3}$|^[1-9]\d{5}[1-9]\d{3}((0\d)|(1[0-2]))(([0|1|2]\d)|3[0-1])((\d{4})|\d{3}[Xx])$/;

// 正则-一年12个月
var RE_MONTH = /^(0?[1-9]|1[0-2])$/;

// 正则-一月的31天
var RE_MONTH_DAY = /^((0?[1-9])|((1|2)[0-9])|30|31)$/;

// 正则-UUID
var RE_UUID = /^[0-9a-zA-Z]{8}-([0-9a-zA-Z]{4}-){3}[0-9a-zA-Z]{12}$/;

// 正则-双字节字符
var RE_DB_CHAR = /^[^x00-xff]*$/;

// 正则-汉字
var RE_CN_CHAR = /^[\u4e00-\u9fa5]*$/;

// 正则-Base64
var RE_BASE64 = /^((data:.*;)?base64,)?([0-9a-zA-Z+/]{4})*(([0-9a-zA-Z+/]{2}==)|([0-9a-zA-Z+/]{3}=))?$/;

// 正则-IPv4
var RE_IPV4 = /^((25[0-5]|2[0-4]\d|[01]?\d\d?)\.){3}(25[0-5]|2[0-4]\d|[01]?\d\d?)$/;

// 正则-域名
var RE_DOMAIN = /^([0-9a-z_!~*'()-]+\.)*([0-9a-z][0-9a-z-]{0,61})?[0-9a-z]\.[a-z]{2,6}$/;

// 正则-端口
var RE_PROT = /^[1-9][0-9]{0,3}|[1-5][0-9]{4}|6[0-4][0-9]{3}|65[0-4][0-9]{2}|655[0-2][0-9]|6553[0-5]$/;

// 正则-URL链接后缀
var RE_URL_SUFFIX = /^[-0-9a-zA-Z+&@#/%=~_|?!:,.;]*$/;

// 正则-IP或域名链接
var RE_URL = /^(https?:)?(\/\/)?(((25[0-5]|2[0-4]\d|[01]?\d\d?)\.){3}(25[0-5]|2[0-4]\d|[01]?\d\d?)|([0-9a-z_!~*'()-]+\.)*([0-9a-z][0-9a-z-]{0,61})?[0-9a-z]\.[a-z]{2,6})(:[0-9]{1-6})?([-0-9a-zA-Z+&@#/%=~_|?!:,.;]*)?$/;

// 正则-IP URL链接
var RE_URL_IP = /^(https?:)?\/\/((25[0-5]|2[0-4]\d|[01]?\d\d?)\.){3}(25[0-5]|2[0-4]\d|[01]?\d\d?)(:[0-9]{1-6})?([-0-9a-zA-Z+&@#/%=~_|?!:,.;]*)?$`/;

// 正则-URL域名链接
var RE_URL_DOMAIN = /^(https?:)?\/\/([0-9a-z_!~*'()-]+\.)*([0-9a-z][0-9a-z-]{0,61})?[0-9a-z]\.[a-z]{2,6}(:[0-9]{1-6})?([-0-9a-zA-Z+&@#/%=~_|?!:,.;]*)?$`/;

exports.BUTILS_CACHE = BUTILS_CACHE;
exports.CACHED_SCRIPTS = CACHED_SCRIPTS;
exports.CACHED_STYLES = CACHED_STYLES;
exports.DAY_TIME = DAY_TIME;
exports.DAY_TIME_1 = DAY_TIME_1;
exports.HTML_ESCAPES = HTML_ESCAPES;
exports.HTML_UNESCAPES = HTML_UNESCAPES;
exports.ID_CHAR_SET = ID_CHAR_SET;
exports.RE_BASE64 = RE_BASE64;
exports.RE_CN_CHAR = RE_CN_CHAR;
exports.RE_DB_CHAR = RE_DB_CHAR;
exports.RE_DOMAIN = RE_DOMAIN;
exports.RE_EMAIL = RE_EMAIL;
exports.RE_ID_CARD = RE_ID_CARD;
exports.RE_ID_CARD_V1 = RE_ID_CARD_V1;
exports.RE_ID_CARD_V2 = RE_ID_CARD_V2;
exports.RE_IPV4 = RE_IPV4;
exports.RE_MONTH = RE_MONTH;
exports.RE_MONTH_DAY = RE_MONTH_DAY;
exports.RE_PHONE = RE_PHONE;
exports.RE_PROT = RE_PROT;
exports.RE_QQ = RE_QQ;
exports.RE_TEL = RE_TEL;
exports.RE_URL = RE_URL;
exports.RE_URL_DOMAIN = RE_URL_DOMAIN;
exports.RE_URL_IP = RE_URL_IP;
exports.RE_URL_SUFFIX = RE_URL_SUFFIX;
exports.RE_UUID = RE_UUID;
exports.RE_ZIP_CODE = RE_ZIP_CODE;
