// @bilibili/b-utils-core/lib/index.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

function _interopDefault (ex) { return (ex && (typeof ex === 'object') && 'default' in ex) ? ex['default'] : ex; }

var consts = require('./consts');
var version = _interopDefault(require('./version'));
var isPhone = _interopDefault(require('./isPhone'));
var isEmail = _interopDefault(require('./isEmail'));
var isIdCard = _interopDefault(require('./isIdCard'));
var isUrl = _interopDefault(require('./isUrl'));
var isIpv4 = _interopDefault(require('./isIpv4'));
var isCnChar = _interopDefault(require('./isCnChar'));
var isDbChar = _interopDefault(require('./isDbChar'));
var isBase64 = _interopDefault(require('./isBase64'));
var isArr = _interopDefault(require('./isArr'));
var isObj = _interopDefault(require('./isObj'));
var isFun = _interopDefault(require('./isFun'));
var isStr = _interopDefault(require('./isStr'));
var isBol = _interopDefault(require('./isBol'));
var isNum = _interopDefault(require('./isNum'));
var isNul = _interopDefault(require('./isNul'));
var isUdf = _interopDefault(require('./isUdf'));
var isUdfNul = _interopDefault(require('./isUdfNul'));
var toRowType = _interopDefault(require('./toRowType'));
var array = _interopDefault(require('./array'));
var toArray = _interopDefault(require('./toArray'));
var tulpeArray = _interopDefault(require('./tulpeArray'));
var unique = _interopDefault(require('./unique'));
var shuffle = _interopDefault(require('./shuffle'));
var flatten = _interopDefault(require('./flatten'));
var includes = _interopDefault(require('./includes'));
var findIndex = _interopDefault(require('./findIndex'));
var get = _interopDefault(require('./get'));
var pick = _interopDefault(require('./pick'));
var omit = _interopDefault(require('./omit'));
var deepCopy = _interopDefault(require('./deepCopy'));
var isDeepEqual = _interopDefault(require('./isDeepEqual'));
var pretty = _interopDefault(require('./pretty'));
var debounce = _interopDefault(require('./debounce'));
var throttle = _interopDefault(require('./throttle'));
var randomId = _interopDefault(require('./randomId'));
var randomNum = _interopDefault(require('./randomNum'));
var randomColor = _interopDefault(require('./randomColor'));
var getStringHash = _interopDefault(require('./getStringHash'));
var compareVersion = _interopDefault(require('./compareVersion'));
var parseQueryString = _interopDefault(require('./parseQueryString'));
var parseQs = _interopDefault(require('./parseQs'));
var stringifyQueryString = _interopDefault(require('./stringifyQueryString'));
var stringifyQs = _interopDefault(require('./stringifyQs'));
var escapeHtml = _interopDefault(require('./escapeHtml'));
var encodeHtml = _interopDefault(require('./encodeHtml'));
var unescapeHtml = _interopDefault(require('./unescapeHtml'));
var decodeHtml = _interopDefault(require('./decodeHtml'));
var formatNum = _interopDefault(require('./formatNum'));
var formatMoney = _interopDefault(require('./formatMoney'));
var cutString = _interopDefault(require('./cutString'));
var maskString = _interopDefault(require('./maskString'));
var capitalize = _interopDefault(require('./capitalize'));
var outOfNumShowPlus = _interopDefault(require('./outOfNumShowPlus'));
var trimReLimits = _interopDefault(require('./trimReLimits'));
var CountDown = _interopDefault(require('./CountDown'));
var formatToDate = _interopDefault(require('./formatToDate'));
var formatDate = _interopDefault(require('./formatDate'));
var formatDateTime = _interopDefault(require('./formatDateTime'));
var formatDuration = _interopDefault(require('./formatDuration'));
var formatTime = _interopDefault(require('./formatTime'));
var formatPassedTime = _interopDefault(require('./formatPassedTime'));
var getAgeByBirth = _interopDefault(require('./getAgeByBirth'));
var getAgeByIdCard = _interopDefault(require('./getAgeByIdCard'));
var getDayStartTime = _interopDefault(require('./getDayStartTime'));
var getDayEndTime = _interopDefault(require('./getDayEndTime'));
var getDayOfYear = _interopDefault(require('./getDayOfYear'));
var getWeekOfYear = _interopDefault(require('./getWeekOfYear'));
var getMonthDays = _interopDefault(require('./getMonthDays'));
var getDaysBetween = _interopDefault(require('./getDaysBetween'));
var isLeapYear = _interopDefault(require('./isLeapYear'));
var PageParams = _interopDefault(require('./PageParams'));
var createHttp = _interopDefault(require('./createHttp'));
var toHttps = _interopDefault(require('./toHttps'));
var urlToHttps = _interopDefault(require('./urlToHttps'));
var trimHttp = _interopDefault(require('./trimHttp'));
var urlTrimHttp = _interopDefault(require('./urlTrimHttp'));
var isSupportWebp = _interopDefault(require('./isSupportWebp'));
var formatImg = _interopDefault(require('./formatImg'));
var formatImgSrc = _interopDefault(require('./formatImgSrc'));
var onImagesLoad = _interopDefault(require('./onImagesLoad'));
var getCookie = _interopDefault(require('./getCookie'));
var setCookie = _interopDefault(require('./setCookie'));
var removeCookie = _interopDefault(require('./removeCookie'));
var getCache = _interopDefault(require('./getCache'));
var setCache = _interopDefault(require('./setCache'));
var removeCache = _interopDefault(require('./removeCache'));
var getLocalStorage = _interopDefault(require('./getLocalStorage'));
var setLocalStorage = _interopDefault(require('./setLocalStorage'));
var removeLocalStorage = _interopDefault(require('./removeLocalStorage'));
var getSessionStorage = _interopDefault(require('./getSessionStorage'));
var setSessionStorage = _interopDefault(require('./setSessionStorage'));
var removeSessionStorage = _interopDefault(require('./removeSessionStorage'));
var isMobile = _interopDefault(require('./isMobile'));
var isDesktop = _interopDefault(require('./isDesktop'));
var isIos = _interopDefault(require('./isIos'));
var isAndroid = _interopDefault(require('./isAndroid'));
var isSSR = _interopDefault(require('./isSSR'));
var isInBrowser = _interopDefault(require('./isInBrowser'));
var isZhTw = _interopDefault(require('./isZhTw'));
var isTouchScreen = _interopDefault(require('./isTouchScreen'));
var getBrowserInfo = _interopDefault(require('./getBrowserInfo'));
var getPlatformInfo = _interopDefault(require('./getPlatformInfo'));
var getPerformanceInfo = _interopDefault(require('./getPerformanceInfo'));
var copyText = _interopDefault(require('./copyText'));
var loadScript = _interopDefault(require('./loadScript'));
var insertScript = _interopDefault(require('./insertScript'));
var loadCss = _interopDefault(require('./loadCss'));
var insertCss = _interopDefault(require('./insertCss'));
var isFullscreen = _interopDefault(require('./isFullscreen'));
var requestFullscreen = _interopDefault(require('./requestFullscreen'));
var exitFullscreen = _interopDefault(require('./exitFullscreen'));
var scrollTo = _interopDefault(require('./scrollTo'));
var scrollToTop = _interopDefault(require('./scrollToTop'));
var scrollToBottom = _interopDefault(require('./scrollToBottom'));
var getScrollPosition = _interopDefault(require('./getScrollPosition'));
var getClientSize = _interopDefault(require('./getClientSize'));
var getPageSize = _interopDefault(require('./getPageSize'));
var getVideoSizeRatio = _interopDefault(require('./getVideoSizeRatio'));
var setToFavorite = _interopDefault(require('./setToFavorite'));
var requestAnimationFrame = _interopDefault(require('./requestAnimationFrame'));
var getScreenPosition = _interopDefault(require('./getScreenPosition'));
var openWindow = _interopDefault(require('./openWindow'));
var onWindowResize = _interopDefault(require('./onWindowResize'));
var offWindowResize = _interopDefault(require('./offWindowResize'));
var dispatchCustomEvent = _interopDefault(require('./dispatchCustomEvent'));
var isHTMLElement = _interopDefault(require('./isHTMLElement'));
var isElementMatch = _interopDefault(require('./isElementMatch'));
var getClosestElement = _interopDefault(require('./getClosestElement'));
var getMatchedWrapElement = _interopDefault(require('./getMatchedWrapElement'));
var removeDomElement = _interopDefault(require('./removeDomElement'));
var setCursorPosition = _interopDefault(require('./setCursorPosition'));
var hasClass = _interopDefault(require('./hasClass'));
var addClass = _interopDefault(require('./addClass'));
var removeClass = _interopDefault(require('./removeClass'));
var base64ToBlob = _interopDefault(require('./base64ToBlob'));
var base64ToFile = _interopDefault(require('./base64ToFile'));
var base64ToUnit8Array = _interopDefault(require('./base64ToUnit8Array'));
var canvasVoidBlur = _interopDefault(require('./canvasVoidBlur'));
var downloadFile = _interopDefault(require('./downloadFile'));
var fileOrBlobToBase64 = _interopDefault(require('./fileOrBlobToBase64'));
var imageToBase64 = _interopDefault(require('./imageToBase64'));
var allCustomReport = _interopDefault(require('./allCustomReport'));

// 常量配置
var index = {
  consts: consts,
  version: version,
  isPhone: isPhone,
  isEmail: isEmail,
  isIdCard: isIdCard,
  isUrl: isUrl,
  isIpv4: isIpv4,
  isCnChar: isCnChar,
  isDbChar: isDbChar,
  isBase64: isBase64,
  isArr: isArr,
  isObj: isObj,
  isFun: isFun,
  isStr: isStr,
  isBol: isBol,
  isNum: isNum,
  isNul: isNul,
  isUdf: isUdf,
  isUdfNul: isUdfNul,
  toRowType: toRowType,
  array: array,
  toArray: toArray,
  tulpeArray: tulpeArray,
  unique: unique,
  shuffle: shuffle,
  flatten: flatten,
  includes: includes,
  findIndex: findIndex,
  get: get,
  pick: pick,
  omit: omit,
  deepCopy: deepCopy,
  isDeepEqual: isDeepEqual,
  pretty: pretty,
  debounce: debounce,
  throttle: throttle,
  randomId: randomId,
  randomNum: randomNum,
  randomColor: randomColor,
  getStringHash: getStringHash,
  compareVersion: compareVersion,
  parseQueryString: parseQueryString,
  parseQs: parseQs,
  stringifyQueryString: stringifyQueryString,
  stringifyQs: stringifyQs,
  escapeHtml: escapeHtml,
  encodeHtml: encodeHtml,
  unescapeHtml: unescapeHtml,
  decodeHtml: decodeHtml,
  formatNum: formatNum,
  formatMoney: formatMoney,
  cutString: cutString,
  maskString: maskString,
  capitalize: capitalize,
  outOfNumShowPlus: outOfNumShowPlus,
  trimReLimits: trimReLimits,
  CountDown: CountDown,
  formatToDate: formatToDate,
  formatDate: formatDate,
  formatDateTime: formatDateTime,
  formatDuration: formatDuration,
  formatTime: formatTime,
  formatPassedTime: formatPassedTime,
  getAgeByBirth: getAgeByBirth,
  getAgeByIdCard: getAgeByIdCard,
  getDayStartTime: getDayStartTime,
  getDayEndTime: getDayEndTime,
  getDayOfYear: getDayOfYear,
  getWeekOfYear: getWeekOfYear,
  getMonthDays: getMonthDays,
  getDaysBetween: getDaysBetween,
  isLeapYear: isLeapYear,
  PageParams: PageParams,
  createHttp: createHttp,
  toHttps: toHttps,
  urlToHttps: urlToHttps,
  trimHttp: trimHttp,
  urlTrimHttp: urlTrimHttp,
  isSupportWebp: isSupportWebp,
  formatImg: formatImg,
  formatImgSrc: formatImgSrc,
  onImagesLoad: onImagesLoad,
  getCookie: getCookie,
  setCookie: setCookie,
  removeCookie: removeCookie,
  getCache: getCache,
  setCache: setCache,
  removeCache: removeCache,
  getLocalStorage: getLocalStorage,
  setLocalStorage: setLocalStorage,
  removeLocalStorage: removeLocalStorage,
  getSessionStorage: getSessionStorage,
  setSessionStorage: setSessionStorage,
  removeSessionStorage: removeSessionStorage,
  isMobile: isMobile,
  isDesktop: isDesktop,
  isIos: isIos,
  isAndroid: isAndroid,
  isSSR: isSSR,
  isInBrowser: isInBrowser,
  isZhTw: isZhTw,
  isTouchScreen: isTouchScreen,
  getBrowserInfo: getBrowserInfo,
  getPlatformInfo: getPlatformInfo,
  getPerformanceInfo: getPerformanceInfo,
  copyText: copyText,
  loadScript: loadScript,
  insertScript: insertScript,
  loadCss: loadCss,
  insertCss: insertCss,
  isFullscreen: isFullscreen,
  requestFullscreen: requestFullscreen,
  exitFullscreen: exitFullscreen,
  scrollTo: scrollTo,
  scrollToTop: scrollToTop,
  scrollToBottom: scrollToBottom,
  getScrollPosition: getScrollPosition,
  getClientSize: getClientSize,
  getPageSize: getPageSize,
  getVideoSizeRatio: getVideoSizeRatio,
  setToFavorite: setToFavorite,
  requestAnimationFrame: requestAnimationFrame,
  getScreenPosition: getScreenPosition,
  openWindow: openWindow,
  onWindowResize: onWindowResize,
  offWindowResize: offWindowResize,
  dispatchCustomEvent: dispatchCustomEvent,
  isHTMLElement: isHTMLElement,
  isElementMatch: isElementMatch,
  getClosestElement: getClosestElement,
  getMatchedWrapElement: getMatchedWrapElement,
  removeDomElement: removeDomElement,
  setCursorPosition: setCursorPosition,
  hasClass: hasClass,
  addClass: addClass,
  removeClass: removeClass,
  base64ToBlob: base64ToBlob,
  base64ToFile: base64ToFile,
  base64ToUnit8Array: base64ToUnit8Array,
  canvasVoidBlur: canvasVoidBlur,
  downloadFile: downloadFile,
  fileOrBlobToBase64: fileOrBlobToBase64,
  imageToBase64: imageToBase64,
  allCustomReport: allCustomReport,
};

exports.consts = consts;
exports.version = version;
exports.isPhone = isPhone;
exports.isEmail = isEmail;
exports.isIdCard = isIdCard;
exports.isUrl = isUrl;
exports.isIpv4 = isIpv4;
exports.isCnChar = isCnChar;
exports.isDbChar = isDbChar;
exports.isBase64 = isBase64;
exports.isArr = isArr;
exports.isObj = isObj;
exports.isFun = isFun;
exports.isStr = isStr;
exports.isBol = isBol;
exports.isNum = isNum;
exports.isNul = isNul;
exports.isUdf = isUdf;
exports.isUdfNul = isUdfNul;
exports.toRowType = toRowType;
exports.array = array;
exports.toArray = toArray;
exports.tulpeArray = tulpeArray;
exports.unique = unique;
exports.shuffle = shuffle;
exports.flatten = flatten;
exports.includes = includes;
exports.findIndex = findIndex;
exports.get = get;
exports.pick = pick;
exports.omit = omit;
exports.deepCopy = deepCopy;
exports.isDeepEqual = isDeepEqual;
exports.pretty = pretty;
exports.debounce = debounce;
exports.throttle = throttle;
exports.randomId = randomId;
exports.randomNum = randomNum;
exports.randomColor = randomColor;
exports.getStringHash = getStringHash;
exports.compareVersion = compareVersion;
exports.parseQueryString = parseQueryString;
exports.parseQs = parseQs;
exports.stringifyQueryString = stringifyQueryString;
exports.stringifyQs = stringifyQs;
exports.escapeHtml = escapeHtml;
exports.encodeHtml = encodeHtml;
exports.unescapeHtml = unescapeHtml;
exports.decodeHtml = decodeHtml;
exports.formatNum = formatNum;
exports.formatMoney = formatMoney;
exports.cutString = cutString;
exports.maskString = maskString;
exports.capitalize = capitalize;
exports.outOfNumShowPlus = outOfNumShowPlus;
exports.trimReLimits = trimReLimits;
exports.CountDown = CountDown;
exports.formatToDate = formatToDate;
exports.formatDate = formatDate;
exports.formatDateTime = formatDateTime;
exports.formatDuration = formatDuration;
exports.formatTime = formatTime;
exports.formatPassedTime = formatPassedTime;
exports.getAgeByBirth = getAgeByBirth;
exports.getAgeByIdCard = getAgeByIdCard;
exports.getDayStartTime = getDayStartTime;
exports.getDayEndTime = getDayEndTime;
exports.getDayOfYear = getDayOfYear;
exports.getWeekOfYear = getWeekOfYear;
exports.getMonthDays = getMonthDays;
exports.getDaysBetween = getDaysBetween;
exports.isLeapYear = isLeapYear;
exports.PageParams = PageParams;
exports.createHttp = createHttp;
exports.toHttps = toHttps;
exports.urlToHttps = urlToHttps;
exports.trimHttp = trimHttp;
exports.urlTrimHttp = urlTrimHttp;
exports.isSupportWebp = isSupportWebp;
exports.formatImg = formatImg;
exports.formatImgSrc = formatImgSrc;
exports.onImagesLoad = onImagesLoad;
exports.getCookie = getCookie;
exports.setCookie = setCookie;
exports.removeCookie = removeCookie;
exports.getCache = getCache;
exports.setCache = setCache;
exports.removeCache = removeCache;
exports.getLocalStorage = getLocalStorage;
exports.setLocalStorage = setLocalStorage;
exports.removeLocalStorage = removeLocalStorage;
exports.getSessionStorage = getSessionStorage;
exports.setSessionStorage = setSessionStorage;
exports.removeSessionStorage = removeSessionStorage;
exports.isMobile = isMobile;
exports.isDesktop = isDesktop;
exports.isIos = isIos;
exports.isAndroid = isAndroid;
exports.isSSR = isSSR;
exports.isInBrowser = isInBrowser;
exports.isZhTw = isZhTw;
exports.isTouchScreen = isTouchScreen;
exports.getBrowserInfo = getBrowserInfo;
exports.getPlatformInfo = getPlatformInfo;
exports.getPerformanceInfo = getPerformanceInfo;
exports.copyText = copyText;
exports.loadScript = loadScript;
exports.insertScript = insertScript;
exports.loadCss = loadCss;
exports.insertCss = insertCss;
exports.isFullscreen = isFullscreen;
exports.requestFullscreen = requestFullscreen;
exports.exitFullscreen = exitFullscreen;
exports.scrollTo = scrollTo;
exports.scrollToTop = scrollToTop;
exports.scrollToBottom = scrollToBottom;
exports.getScrollPosition = getScrollPosition;
exports.getClientSize = getClientSize;
exports.getPageSize = getPageSize;
exports.getVideoSizeRatio = getVideoSizeRatio;
exports.setToFavorite = setToFavorite;
exports.requestAnimationFrame = requestAnimationFrame;
exports.getScreenPosition = getScreenPosition;
exports.openWindow = openWindow;
exports.onWindowResize = onWindowResize;
exports.offWindowResize = offWindowResize;
exports.dispatchCustomEvent = dispatchCustomEvent;
exports.isHTMLElement = isHTMLElement;
exports.isElementMatch = isElementMatch;
exports.getClosestElement = getClosestElement;
exports.getMatchedWrapElement = getMatchedWrapElement;
exports.removeDomElement = removeDomElement;
exports.setCursorPosition = setCursorPosition;
exports.hasClass = hasClass;
exports.addClass = addClass;
exports.removeClass = removeClass;
exports.base64ToBlob = base64ToBlob;
exports.base64ToFile = base64ToFile;
exports.base64ToUnit8Array = base64ToUnit8Array;
exports.canvasVoidBlur = canvasVoidBlur;
exports.downloadFile = downloadFile;
exports.fileOrBlobToBase64 = fileOrBlobToBase64;
exports.imageToBase64 = imageToBase64;
exports.allCustomReport = allCustomReport;
exports.default = index;
